.ado/templates/react-native-init.yml

```diff
@@ -114,34 +108,26 @@ steps:
         -Cleanup:$true
     condition: and(succeeded(), ${{ parameters.installVsComponents }})
 
-  - task: VSBuild@1
-    displayName: VSBuild - testcli
+  # Work around issue of parameters not getting expanded in conditions properly
+  - powershell: |
+      Write-Host "##vso[task.setvariable variable=localConfig]${{ parameters.configuration}}"
+
+  - task: CmdLine@2
+    displayName: Build project (Release)
     inputs:
-      solution: $(Agent.BuildDirectory)\testcli\windows\testcli.sln
-      vsVersion: $(MSBuildVersion) # Optional. Options: latest, 16.0, 15.0, 14.0, 12.0, 4.0
-      msbuildArchitecture: $(MSBuildArchitecture) # Optional. Options: x86, x64
-      platform: ${{ parameters.platform }} # Optional
-      configuration: ${{ parameters.configuration }} # Optional
-      clean: false # Optional
-      maximumCpuCount: false # Optional
-      restoreNugetPackages: false # Optional
-      createLogFile: true
-      logFileVerbosity: detailed
-      msbuildArgs:
-        /p:PreferredToolArchitecture=$(MSBuildPreferredToolArchitecture)
-        /p:PlatformToolset=$(MSBuildPlatformToolset)
-        /p:AppxGeneratePrisForPortableLibrariesEnabled=false
-
-  - task: PublishBuildArtifacts@1
-    condition:  succeededOrFailed()
-    timeoutInMinutes: 10
+      script: npx --no-install react-native run-windows --arch ${{ parameters.platform }} --no-launch --no-deploy --release
+      workingDirectory: $(Agent.BuildDirectory)\testcli
+    condition: and(succeeded(), eq('Release', variables['localConfig']))
+
+  - task: CmdLine@2
+    displayName: Build project (Debug)
     inputs:
-      pathtoPublish: $(Agent.BuildDirectory)/testcli/windows/testcli.sln.log
-      artifactName: Detailed Logs for testcli.sln ${{ parameters.platform }} ${{ parameters.configuration }} ${{ parameters.experimentalNugetDependency }}
-      publishLocation: 'Container'
+      script: npx --no-install react-native run-windows --arch ${{ parameters.platform }} --no-launch --no-deploy
```

Comment:
If it's not to noisy, should we run these with verbose flags to get better logs?